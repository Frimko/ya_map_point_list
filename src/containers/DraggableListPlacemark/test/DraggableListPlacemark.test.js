import React from 'react';
import { shallow, mount } from 'enzyme';
import { List, Icon, Input, Form } from 'antd';

import { DraggableListPlacemark } from '../index';

describe('test DraggableListPlacemark component', () => {


  const props = {
    placemarks: [
      {
        id: 111,
        coordinates: [
          55.796702547976345,
          37.782572021483865,
        ],
        name: 'Это метка1',
      },
      {
        id: 222,
        coordinates: [
          55.67938453142209,
          37.47520507812455,
        ],
        name: 'Это метка2',
      },
      {
        id: 333,
        coordinates: [
          55.84690514362938,
          37.62353544921852,
        ],
        name: 'Это метка3',
      },
      {
        id: 444,
        coordinates: [
          55.68559165521214,
          37.764969482421186,
        ],
        name: 'Это метка4',
      },
      {
        id: 555,
        coordinates: [
          55.79328258155638,
          37.46009887695268,
        ],
        name: 'Это метка5',
      },
      {
        id: 666,
        coordinates: [
          55.795603556125,
          37.78007568359225,
        ],
        name: 'Это метка6',
      },
    ],
    setNewMark: jest.fn().mockName('setNewMark'),
    deleteMark: jest.fn().mockName('deleteMark'),
    changeSortMark: jest.fn().mockName('changeSortMark'),
  };
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <DraggableListPlacemark {...props}/>,
    );
  });

  test('snapshot', () => {
    expect(shallow(
      <DraggableListPlacemark {...props}/>,
    )).toMatchSnapshot();
  });

  test('delete item', () => {
    wrapper.find(List.Item).find(Icon).forEach((node, index) => {
      node.simulate('click', { preventDefault: () => ({}) });
      expect(props.deleteMark).toBeCalledWith(props.placemarks[index].id);
    });
  });

  test('new item submit', () => {
    const text = 'test';
    const input = wrapper.find(Input);
    input.simulate('change', { target: { value: text } });
    wrapper.find(Form).simulate('submit', { preventDefault: () => ({}) });
    expect(props.setNewMark).toBeCalledWith(text);
  });

  test('new item error', () => {
    const input = wrapper.find(Input);
    input.simulate('change', { target: { value: 1 } });
    wrapper.find(Form).simulate('submit', { preventDefault: () => ({}) });
    expect(props.setNewMark).not.toBeCalledWith(1);
  });

  test('drag item', () => {
    // к сожалению через энзим протестировать не выйдет, а юзать react-dom/test-utils то еще приключение
  });
});
