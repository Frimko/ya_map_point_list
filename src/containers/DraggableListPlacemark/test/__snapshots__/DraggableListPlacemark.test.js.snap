// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DraggableListPlacemark component snapshot 1`] = `
<Styled(div)>
  <sortableList
    axis="y"
    deleteMark={[Function]}
    distance={5}
    getHelperDimensions={[Function]}
    helperClass="css-16v6gnb"
    hideSortableGhost={true}
    items={
      Array [
        Object {
          "coordinates": Array [
            55.796702547976345,
            37.782572021483865,
          ],
          "id": 111,
          "name": "Это метка1",
        },
        Object {
          "coordinates": Array [
            55.67938453142209,
            37.47520507812455,
          ],
          "id": 222,
          "name": "Это метка2",
        },
        Object {
          "coordinates": Array [
            55.84690514362938,
            37.62353544921852,
          ],
          "id": 333,
          "name": "Это метка3",
        },
        Object {
          "coordinates": Array [
            55.68559165521214,
            37.764969482421186,
          ],
          "id": 444,
          "name": "Это метка4",
        },
        Object {
          "coordinates": Array [
            55.79328258155638,
            37.46009887695268,
          ],
          "id": 555,
          "name": "Это метка5",
        },
        Object {
          "coordinates": Array [
            55.795603556125,
            37.78007568359225,
          ],
          "id": 666,
          "name": "Это метка6",
        },
      ]
    }
    lockAxis="y"
    lockOffset="50%"
    lockToContainerEdges={false}
    onSortEnd={[Function]}
    pressDelay={0}
    pressThreshold={5}
    shouldCancelStart={[Function]}
    submit={[Function]}
    transitionDuration={300}
    useWindowAsScrollContainer={false}
  />
</Styled(div)>
`;
